tests/failpoints/cases/test_async_io.rs

```diff
@@ -137,6 +140,93 @@ fn test_async_io_apply_conf_change_without_leader_persist() {
     eventually_get_equal(&cluster.get_engine(1), b"k1", b"v3");
 }
 
+#[test]
+fn test_async_io_apply_before_persist_apply_snapshot() {
+    let mut cluster = new_node_cluster(0, 3);
+    cluster.cfg.raft_store.store_io_pool_size = 1;
```

Comment:
In test I prefer to use multiple io thread instead of 1 to expose more potential corner cases.